### PR TITLE
update cosmiconfig, mjs config support, reloading bugfix

### PR DESCRIPTION
**Pull request by phryneas**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -8,8 +8,10 @@
       "request": "launch",
       "preLaunchTask": "npm: watch",
       "runtimeExecutable": "${execPath}",
-      "args": ["--extensionDevelopmentPath=${workspaceRoot}"],
-      "stopOnEntry": false,
+      "args": [
+        "--extensionDevelopmentPath=${workspaceRoot}",
+        "--disable-extensions"
```
Comment (phryneas) [inline .vscode/launch.json]: This disables all other extensions during debugging.

File: src/language-server/config/loadConfig.ts [hunk 2]
```diff
@@ -22,20 +20,11 @@ const defaultFileNames = [
...
 ];
 const envFileNames = [".env", ".env.local"];
 
-const loaders = {
-  // XXX improve types for config
-  ".json": (cosmiconfig as any).loadJson as LoaderEntry,
-  ".js": (cosmiconfig as any).loadJs as LoaderEntry,
-  ".cjs": (cosmiconfig as any).loadJs as LoaderEntry,
-  ".ts": {
-    async: TypeScriptLoader,
```
Comment (phryneas) [inline src/language-server/config/loadConfig.ts:35]: TypeScript is supported per default now.

**Comment by phryneas** [comment]
And ready for review too :)

File: src/language-server/workspace.ts [hunk 3]
```diff
@@ -180,7 +180,7 @@ export class GraphQLWorkspace {
 
   async reloadProjectForConfig(configUri: DocumentUri) {
     const configPath = dirname(URI.parse(configUri).fsPath);
-
+    console.log("reloadProjectForConfig");
```
Comment (jerelmiller) [inline src/language-server/workspace.ts]:
```suggestion
```

**Review by jerelmiller — APPROVED** [review]
💯

**Pull request merged**